jetty-websocket/websocket-servlet/src/main/java/org/eclipse/jetty/websocket/servlet/WebSocketServlet.java

```diff
@@ -53,167 +51,242 @@
  *     &#064;Override
  *     public void configure(WebSocketServletFactory factory)
  *     {
- *         // set a 10 second idle timeout
- *         factory.getPolicy().setIdleTimeout(10000);
- *         // register my socket
- *         factory.register(MyEchoSocket.class);
+ *       factory.setDefaultMaxFrameSize(4096);
+ *       factory.addMapping(factory.parsePathSpec("/"), (req,res)-&gt;new EchoSocket());
```

Comment:
the shorthand `factory.register(Class)` needs to still be here for backwards compat reasons.

Reply:
hmmm that has long gone, so perhaps not a subject for this PR.  Happy to add it back as a deprecated method, but can you explain exactly what it does? `registeredSocketClasses` no longer exists as a concept?

Reply:
I've opened #3256 to handle this separately